Assets/MixedRealityToolkit.Services/BoundarySystem/MixedRealityBoundarySystem.cs

```diff
@@ -289,6 +289,9 @@ private GameObject BoundaryVisualizationParent
         /// </summary>
         private int ignoreRaycastLayerValue = 2;
 
+        /// <inheritdoc/>
+        public MixedRealityBoundaryVisualizationProfile BoundaryVisualizationProfile => ConfigurationProfile as MixedRealityBoundaryVisualizationProfile;
```

Comment:
Should we cache these `as` calls so we aren't casting every time?

Reply:
good question... we _could_

Reply:
here's the new pattern:

```
        public MixedRealityBoundaryVisualizationProfile BoundaryVisualizationProfile
        {
            get
            {
                if (boundaryVisualizationProfile == null)
                {
                    boundaryVisualizationProfile = ConfigurationProfile as MixedRealityBoundaryVisualizationProfile;
                }
                return boundaryVisualizationProfile;
            }
        }
```